CRM/Contact/BAO/Relationship.php

```diff
@@ -1748,8 +1740,8 @@ public static function relatedMemberships($contactId, &$params, $ids, $action =
             if ($action & CRM_Core_Action::UPDATE) {
               //if updated relationship is already related to contact don't delete existing inherited membership
               if (in_array($relTypeId, $relTypeIds
-                ) && !empty($values[$relatedContactId]['memberships']) && !empty($ownerMemIds
-                ) && in_array($membershipValues['owner_membership_id'], $ownerMemIds[$relatedContactId])) {
+                ) && !empty($values[$relatedContactId]['memberships'])
+                && ($membershipInherittedFromContactID !== (int) $membershipValues['contact_id'])) {
```

Comment:
`$membershipInherittedFromContactID` will be`NULL` if there is no [`owner_membership_id`](https://github.com/civicrm/civicrm-core/blob/5ecf0e710f52cbb1bac0514e53fa1fbc8b89a167/CRM/Contact/BAO/Relationship.php#L1679) for the current membership values (it is the parent membership). See [the line](https://github.com/civicrm/civicrm-core/blob/5ecf0e710f52cbb1bac0514e53fa1fbc8b89a167/CRM/Contact/BAO/Relationship.php#L1682) where `$membershipInherittedFromContactID` is defined.

The membership is added in [this line](https://github.com/civicrm/civicrm-core/blob/5ecf0e710f52cbb1bac0514e53fa1fbc8b89a167/CRM/Contact/BAO/Relationship.php#L1759) because `NULL !== int`.